frontend/src/employee-frontend/components/reports/Occupancies.tsx

```diff
@@ -122,9 +122,10 @@ function getDisplayCells(
 ): string[][] {
   return reportRows.map((row) => {
     if (usedValues === 'raw') {
-      const cells = [row.unitName, row.groupName].filter(
-        (cell) => cell !== undefined
-      ) as string[]
+      const cells = [
+        row.unitName,
+        'groupName' in row ? row.groupName : undefined
+      ].filter((cell): cell is string => cell !== undefined)
```

Comment:
Here's a trick to avoid the type annotation and make this a bit shorter:

```
const cells = [
  row.unitName,
  'groupName' in row ? row.groupName : undefined
].flatMap((cell) => cell !== undefined ? cell : [])
```

The applicability of the trick depends on whether the future readers understand that `flatMap` here is actually used for filtering.

On the teoretical side of things, it's actually more correct to use `flatMap`, because most type systems don't allow changing the collection type with `filter`. (In this case the type is changed from `(string | undefined)[]` to `string[]`.)

Reply:
I redid this part of the code and handle this one optional cell in a different way. It also fixes a potential bug where null in the data would lead to a missing cell (= broken table layout)